### PR TITLE
Feature/redirect back or error

### PR DESCRIPTION
**Pull request by martenwallewein**

### CODE REVIEW

File: chrome/popup.js
Comment (JordiSubira) [inline chrome/popup.js:110]: Some times not all domains are shown in the popus, I wonder if it has to do with how and when this checkedDomains set is (re-)created.

File: chrome/popup.js
Comment (JordiSubira) [inline chrome/popup.js]: Remove?

File: chrome/options.js
Comment (JordiSubira) [inline chrome/options.js]: Does this then() clause do anthing useful?

File: chrome/database.js
Comment (JordiSubira) [inline chrome/database.js]: Is this a leftover comment? Remove  it if that's the case

File: chrome/database.js
Comment (JordiSubira) [inline chrome/database.js]: Remove reject, it isn't used.

File: chrome/database.js
Comment (JordiSubira) [inline chrome/database.js]: Remove reject, it isn't used.

File: chrome/database.js
Comment (JordiSubira) [inline chrome/database.js]: Can we briefly explain why do we have this? So that we can change/fix it in the future?

File: chrome/background.js
Comment (JordiSubira) [inline chrome/background.js]: Remove?

File: chrome/background.js
Comment (JordiSubira) [inline chrome/background.js]: Do we need this if we already have the onErrorOccured/onBeforeRedirect hack?

**Review by JordiSubira — CHANGES_REQUESTED** [review]

File: chrome/popup.js
Comment (martenwallewein) [reply chrome/popup.js:110]: I will check this next. So you can keep the PR open until it's fixed or merge it, what you prefer. But a good indication, thanks!

File: chrome/database.js
Comment (martenwallewein) [reply chrome/database.js]: Done

File: chrome/background.js
Comment (martenwallewein) [reply chrome/background.js]: This was quite interesting. I tried to do it in this hack but it didn't work. I wrote a comment why.

File: chrome/popup.js
Comment (martenwallewein) [reply chrome/popup.js:110]: I think this is solved. We should merge this branch before HotNets if we want to reference the code in the paper

File: chrome/database.js
Comment (JordiSubira) [inline chrome/database.js:62]: I tested it out and works fine for me. IMO it is an acceptable solution for now even the persistance logic seems quite complex for what we are doing so far. I think first we need a clear strategy how to proceed with the extension development.

**Review by JordiSubira — APPROVED** [review]

File: chrome/database.js
Comment (martenwallewein) [reply chrome/database.js:62]: I definitely agree to this!

**Pull request merged**
